controllers/configurationpolicy_controller.go

```diff
@@ -648,31 +688,31 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 		// kick off object deletion if configurationPolicy has been deleted
 		if plc.ObjectMeta.DeletionTimestamp != nil {
 			// If the CRD is deleting, don't prune objects
-			crdDeleting, err := r.definitionIsDeleting()
+			cleanupNow, err := r.cleanupImmediately()
```

Comment:
Will this get triggered in hosted mode where only a single hosted cluster is deleted but one other remains? I think `plc.ObjectMeta.DeletionTimestamp != nil` would only be true if the CRD was being deleted.

Reply:
I was assuming the namespace would be deleted when the controller was being uninstalled, but I think you're right and there are situations that wouldn't be true.

This also made me realize a problem with the `cleanupImmediately` implementation where if there was an error getting the deployment, but the CRD is successfully found *not* to be deleting, it decides immediate cleanup is *not* necessary. But like you say, in hosted mode, that's the wrong decision. If it can't get the deployment, it can't decide that immediate cleanup is not necessary.